scripts/linux/cross-platform-rust-build.sh

```diff
@@ -0,0 +1,293 @@
+    echo ""
+    echo "options"
+    echo " --os                      Desired os for build"
+    echo " --arch                    Desired arch for build"
+    echo " --build-path              Desired arch for build"
+    echo " -h, --help                Print this help and exit."
+    exit 1;
+}
+
+###############################################################################
+# Obtain and validate the options supported by this script
+###############################################################################
+process_args()
+{
+    save_next_arg=0
+    for arg in "$@"
+    do
+        if [ $save_next_arg -eq 1 ]; then
+            PACKAGE_OS="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 2 ]; then
+            PACKAGE_ARCH="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 3 ]; then
+            BUILD_PATH="$arg"
+            save_next_arg=0
+        else
+            case "$arg" in
+                "--os" ) save_next_arg=1;;
+                "--arch" ) save_next_arg=2;;
+                "--build-path" ) save_next_arg=3;;
+                "-h" | "--help" ) usage;;
+                * ) usage;;
+            esac
+        fi
+    done
+}
+
+process_args "$@"
+
+[[ -z "$PACKAGE_OS" ]] && { print_error 'OS is a required parameter'; exit 1; }
+[[ -z "$PACKAGE_ARCH" ]] && { print_error 'Arch is a required parameter'; exit 1; }
+[[ -z "$BUILD_PATH" ]] && { print_error 'Build path is a required parameter'; exit 1; }
+
+set -e
+
+# Get directory of running script
+DIR="$(cd "$(dirname "$0")" && pwd)"
+
+BUILD_REPOSITORY_LOCALPATH="$(realpath "${BUILD_REPOSITORY_LOCALPATH:-$DIR/../..}")"
+
+CMAKE_ARGS='-DCMAKE_BUILD_TYPE=Release'
+CMAKE_ARGS="$CMAKE_ARGS -DBUILD_SHARED=On -Drun_unittests=Off -Duse_default_uuid=On -Duse_emulator=Off -Duse_http=Off"
+
+DOCKER_VOLUME_MOUNTS=''
+
+case "$PACKAGE_OS" in
+    'ubuntu18.04')
+        DOCKER_IMAGE='ubuntu:18.04'
+        ;;
+esac
+
+if [ -z "$DOCKER_IMAGE" ]; then
+    echo "Unrecognized target [$PACKAGE_OS.$PACKAGE_ARCH]" >&2
+    exit 1
+fi
+
+case "$PACKAGE_ARCH" in
+    'amd64')
+        RUST_TARGET='x86_64-unknown-linux-musl'
+        ;;
+
+    'arm32v7')
+        RUST_TARGET='armv7-unknown-linux-gnueabihf'
+        ;;
+
+    'aarch64')
+        RUST_TARGET='aarch64-unknown-linux-gnu'
+        ;;
+esac
+
+if [ -n "$RUST_TARGET" ]; then
+    RUST_TARGET_COMMAND="rustup target add $RUST_TARGET &&"
+fi
+
+
+case "$PACKAGE_OS.$PACKAGE_ARCH" in
+    ubuntu18.04.amd64)
+        # The below SETUP was copied from https://github.com/emk/rust-musl-builder/blob/master/Dockerfile.
+        # The beginning of the below setup specifies versions for other dependencies.
+        # Here are the places to check for new releases:
+        #
+        # - https://github.com/rust-lang/mdBook/releases
+        # - https://github.com/EmbarkStudios/cargo-about/releases
+        # - https://github.com/EmbarkStudios/cargo-deny/releases
```

Comment:
We need to find a way to handle these important dependencies since they will get updated. Any ideas? We can elevate to the top of file for clarity, but that will separate it from the alpine logic.

Reply:
We don't need MDBOOK as far as I know, so it can be safely removed. I'm also not sure if we use Cargo About/Deny plugins. @dmolokanov any thoughts?

We also don't need `graphviz` package, as far as I know.

Reply:
Why does the musl-builder install these?

Reply:
I believe musl-builder tends to be a generic builder image that tries to satisfy many popular dependencies. We may not need all of them...